src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -181,12 +189,28 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 
 	// if we have a valid velocity setpoint and the vehicle is demanded to go down but no vertical movement present,
 	// we then can assume that the vehicle hit ground
-	_in_descend = _is_climb_rate_enabled()
-		      && (_vehicle_local_position_setpoint.vz >= land_speed_threshold);
+	if (_is_climb_rate_enabled()) {
+		if (PX4_ISFINITE(_vehicle_local_position_setpoint.vz)) {
+			_in_descend = (_vehicle_local_position_setpoint.vz >= land_speed_threshold);
+		}
```

Comment:
Should there be an `else` here with: `_in_descend = false` ?

Reply:
Ultimately, but at the moment the multicopter position controller is publishing NANs briefly when it first encounters ground contact.

Reply:
I've added a comment with a TODO.